package.json

```diff
@@ -68,64 +70,73 @@
   },
   "devDependencies": {
+    "@babel/eslint-parser": "^7.21.3",
+    "@babel/plugin-proposal-decorators": "^7.21.0",
     "@ember/optional-features": "^2.0.0",
-    "@ember/test-helpers": "^2.4.2",
-    "@embroider/test-setup": "^0.50.2",
+    "@ember/string": "^3.0.1",
+    "@ember/test-helpers": "^3.2.0",
+    "@embroider/test-setup": "^3.0.1",
     "@fortawesome/free-brands-svg-icons": "^6.2.0",
     "@fortawesome/free-regular-svg-icons": "^6.2.0",
     "@fortawesome/free-solid-svg-icons": "^6.2.0",
-    "@glimmer/component": "^1.0.4",
-    "@glimmer/tracking": "^1.0.4",
-    "babel-eslint": "^10.1.0",
+    "@glimmer/component": "^1.1.2",
+    "@glimmer/tracking": "^1.1.2",
     "broccoli-asset-rev": "^3.0.0",
     "broccoli-stew": "^3.0.0",
     "broccoli-viz": "^5.0.0",
-    "ember-cli": "~3.28.2",
-    "ember-cli-dependency-checker": "^3.2.0",
+    "concurrently": "^8.0.1",
+    "ember-cli": "~4.12.0",
+    "ember-cli-dependency-checker": "^3.3.1",
     "ember-cli-inject-live-reload": "^2.1.0",
     "ember-cli-sri": "^2.1.1",
     "ember-cli-terser": "^4.0.2",
-    "ember-disable-prototype-extensions": "^1.1.3",
-    "ember-export-application-global": "^2.0.1",
+    "ember-compatibility-helpers": "^1.2.6",
     "ember-load-initializers": "^2.1.2",
-    "ember-maybe-import-regenerator-for-testing": "^1.0.0",
-    "ember-page-title": "^6.2.2",
-    "ember-qunit": "^5.1.4",
-    "ember-resolver": "^8.0.2",
-    "ember-source": "~3.28.0",
+    "ember-page-title": "^8.0.0",
+    "ember-qunit": "^7.0.0",
+    "ember-resolver": "^11.0.0",
+    "ember-source": "~4.12.0",
     "ember-source-channel-url": "^3.0.0",
-    "ember-template-lint": "^3.6.0",
-    "ember-try": "^1.4.0",
-    "eslint": "^7.32.0",
-    "eslint-config-prettier": "^8.3.0",
-    "eslint-plugin-ember": "^10.5.4",
-    "eslint-plugin-node": "^11.1.0",
-    "eslint-plugin-prettier": "^3.4.1",
-    "eslint-plugin-qunit": "^6.2.0",
-    "husky": "^3.0.5",
+    "ember-template-lint": "^5.7.2",
+    "ember-try": "^2.0.0",
+    "eslint": "^8.37.0",
+    "eslint-config-prettier": "^9.0.0",
+    "eslint-plugin-ember": "^11.10.0",
+    "eslint-plugin-n": "^16.0.1",
+    "eslint-plugin-prettier": "^5.0.0",
+    "eslint-plugin-qunit": "^8.0.0",
+    "husky": "^8.0.3",
     "lint-staged": "^9.2.5",
     "loader.js": "^4.7.0",
     "markdown-toc": "^1.2.0",
-    "npm-run-all": "^4.1.5",
-    "prettier": "^2.3.2",
-    "qunit": "^2.16.0",
-    "qunit-dom": "^1.6.0",
-    "webpack": "^5.66.0"
+    "prettier": "^3.0.2",
+    "qunit": "^2.19.4",
+    "qunit-dom": "^2.0.0",
+    "stylelint": "^15.10.3",
+    "stylelint-config-standard": "^34.0.0",
+    "stylelint-prettier": "^4.0.2",
+    "webpack": "^5.78.0"
+  },
+  "peerDependencies": {
+    "ember-source": "^3.20.0 || ^4.0.0 || >=5.0.0"
   },
   "engines": {
-    "node": "12.* || 14.* || >= 16"
+    "node": "14.* || 16.* || >= 18"
   },
   "ember": {
     "edition": "octane"
   },
+  "overrides": {
+    "ember-source": "$ember-source"
```

Comment:
Something went wrong here I think.

Reply:
@jrjohnson no, this is necessary to fix the peerDependency check for release, beta... tests.

The version of ember-source is in release like `ember-source@5.2.0-release+a1cd26c4`. Npm has problems in this situation. I was already running in the same problem in other projects, which were using `ember-render-modifiers` (https://github.com/emberjs/ember-render-modifiers/issues/64#issuecomment-1114913950)

I know, its not the best solution, but it helps us to fix the tests